From cc9045837fd8690c029b8508ed1bc471b5c3ad5b Mon Sep 17 00:00:00 2001
From: Bryce Harrington <bryce@bryceharrington.org>
Date: Tue, 24 Feb 2009 14:40:13 -0800
Subject: [PATCH] Update bulgarian layouts and add new standard phonetic layout. #20301

On Dec 5, 2007 the Bulgarian government created a new standard phonetic
keyboard layout which differs from the traditional phonetic keyboard
layout.  This patch adds it as a third option, and updates the standard
with minor modifications recommended by the Bulgarian Academy of Science.

Signed-off-by: Bryce Harrington <bryce@bryceharrington.org>
---
 rules/base.xml.in |   10 +++++-
 symbols/bg        |  101 ++++++++++++++++++++++++++++++++++++++++++++++-------
 2 files changed, 97 insertions(+), 14 deletions(-)

diff --git a/rules/base.xml.in b/rules/base.xml.in
index f12390a..ef7a4fb 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -1771,7 +1771,15 @@
         <variant>
           <configItem>
             <name>phonetic</name>
-            <_description>Phonetic</_description>
+            <_description>Traditional Phonetic</_description>
+          </configItem>
+        </variant>
+      </variantList>
+      <variantList>
+        <variant>
+          <configItem>
+            <name>bas_phonetic</name>
+            <_description>Standard Phonetic</_description>
           </configItem>
         </variant>
       </variantList>
diff --git a/symbols/bg b/symbols/bg
index 8c4913c..b990d20 100644
--- a/symbols/bg
+++ b/symbols/bg
@@ -1,9 +1,5 @@
-// $XKeyboardConfig$
-
-// based on a keyboard map:
-//
 //////////////////////////////////////////////////////////////////////////
-// Copyright (C) 1999, 2000 by Anton Zinoviev <anton@lml.bas.bg>
+// Copyright (C) 1999, 2000, 2005 by Anton Zinoviev <anton@lml.bas.bg>
 //
 // This software may be used, modified, copied, distributed, and sold,
 // in both source and binary form provided that the above copyright
@@ -13,14 +9,13 @@
 // use.
 //
 ////////////////////////////////////////////////////////////////////////// 
-// Version 1.6r1
+// Version 1.9-BAN
 
 partial default alphanumeric_keys
 xkb_symbols "bds" {
-// BDS is the Bulgarian Institute for Standardization
-
   name[Group1]= "Bulgaria";
   
+key.type[Group1] = "TWO_LEVEL";
   key <TLDE> {   [ parenleft,         parenright          ]    };
   key <AE01> {   [ 1,                 exclam              ]    };
   key <AE02> {   [ 2,                 question            ]    };
@@ -32,10 +27,11 @@ xkb_symbols "bds" {
   key <AE08> {   [ 8,                 slash               ]    };
   key <AE09> {   [ 9,                 emdash              ]    };
   key <AE10> {   [ 0,                 numerosign          ]    };
-  key <AE11> {   [ minus,             I                   ]    };
-  key <AE12> {   [ period,            V                   ]    };
+  key <AE11> {   [ minus,             dollar              ]    };
+  key <AE12> {   [ period,            EuroSign            ]    };
 
   key <AD01> {   [ comma,             Cyrillic_yeru       ]    };
+key.type[Group1] = "ALPHABETIC";
   key <AD02> {   [ Cyrillic_u,        Cyrillic_U          ]    };
   key <AD03> {   [ Cyrillic_ie,       Cyrillic_IE         ]    };
   key <AD04> {   [ Cyrillic_i,        Cyrillic_I          ]    };
@@ -46,9 +42,11 @@ xkb_symbols "bds" {
   key <AD09> {   [ Cyrillic_de,       Cyrillic_DE         ]    };
   key <AD10> {   [ Cyrillic_ze,       Cyrillic_ZE         ]    };
   key <AD11> {   [ Cyrillic_tse,      Cyrillic_TSE        ]    };
+key.type[Group1] = "TWO_LEVEL";
   key <AD12> {   [ semicolon,         section             ]    };
 
-  key <AC01> {   [ Cyrillic_softsign, Cyrillic_SOFTSIGN   ]    };
+key.type[Group1] = "ALPHABETIC";
+  key <AC01> {   [ Cyrillic_softsign, U045D               ]    };
   key <AC02> {   [ Cyrillic_ya,       Cyrillic_YA         ]    };
   key <AC03> {   [ Cyrillic_a,        Cyrillic_A          ]    };
   key <AC04> {   [ Cyrillic_o,        Cyrillic_O          ]    };
@@ -59,7 +57,6 @@ xkb_symbols "bds" {
   key <AC09> {   [ Cyrillic_ve,       Cyrillic_VE         ]    };
   key <AC10> {   [ Cyrillic_em,       Cyrillic_EM         ]    };
   key <AC11> {   [ Cyrillic_che,      Cyrillic_CHE        ]    };
-  key <BKSL> {   [ apostrophe,        Cyrillic_YERU       ]    };
 
   key <AB01> {   [ Cyrillic_yu,       Cyrillic_YU         ]    };
   key <AB02> {   [ Cyrillic_shorti,   Cyrillic_SHORTI     ]    };
@@ -72,14 +69,22 @@ xkb_symbols "bds" {
   key <AB09> {   [ Cyrillic_el,       Cyrillic_EL         ]    };
   key <AB10> {   [ Cyrillic_be,       Cyrillic_BE         ]    };
     
+key.type[Group1] = "TWO_LEVEL";
+  key <BKSL> {   [ doublelowquotemark,leftdoublequotemark ]    };
+
+  key <LSGT> {   [ Cyrillic_softsign, Cyrillic_SOFTSIGN   ]    };
+
   // End alphanumeric section
   
+  key <KPDL> {   type[Group1] = "KEYPAD",
+                 [ KP_Delete,         KP_Separator        ]    };
+
 };
 
 partial alphanumeric_keys 
 xkb_symbols "phonetic" {
   
-  name[Group1]= "Bulgaria - Phonetic";
+  name[Group1]= "Bulgaria - Traditional Phonetic";
   
   key <TLDE> {   [ Cyrillic_che,      Cyrillic_CHE        ]    };
   key <AE01> {   [ 1,                 exclam              ]    };
@@ -135,3 +140,73 @@ xkb_symbols "phonetic" {
   // End alphanumeric section
 
 };
+
+partial default alphanumeric_keys
+xkb_symbols "bas_phonetic" {
+  name[Group1]= "Bulgarian - Standard Phonetic";
+
+key.type[Group1] = "ALPHABETIC";
+  key <TLDE> {   [ Cyrillic_yu,         Cyrillic_YU          ]    };
+key.type[Group1] = "TWO_LEVEL";
+  key <AE01> {   [ 1,                   exclam               ]    };
+  key <AE02> {   [ 2,                   at                   ]    };
+  key <AE03> {   [ 3,                   numerosign           ]    };
+  key <AE04> {   [ 4,                   dollar               ]    };
+  key <AE05> {   [ 5,                   percent              ]    };
+  key <AE06> {   [ 6,                   EuroSign             ]    };
+  key <AE07> {   [ 7,                   section              ]    };
+  key <AE08> {   [ 8,                   asterisk             ]    };
+  key <AE09> {   [ 9,                   parenleft            ]    };
+  key <AE10> {   [ 0,                   parenright           ]    };
+  key <AE11> {   [ minus,               emdash               ]    };
+  key <AE12> {   [ equal,               plus                 ]    };
+
+key.type[Group1] = "ALPHABETIC";
+  key <LatQ> {   [ Cyrillic_che,        Cyrillic_CHE         ]    };
+  key <LatW> {   [ Cyrillic_sha,        Cyrillic_SHA         ]    };
+  key <LatE> {   [ Cyrillic_ie,         Cyrillic_IE          ]    };
+  key <LatR> {   [ Cyrillic_er,         Cyrillic_ER          ]    };
+  key <LatT> {   [ Cyrillic_te,         Cyrillic_TE          ]    };
+  key <LatY> {   [ Cyrillic_hardsign,   Cyrillic_HARDSIGN    ]    };
+  key <LatU> {   [ Cyrillic_u,          Cyrillic_U           ]    };
+  key <LatI> {   [ Cyrillic_i,          Cyrillic_I           ]    };
+  key <LatO> {   [ Cyrillic_o,          Cyrillic_O           ]    };
+  key <LatP> {   [ Cyrillic_pe,         Cyrillic_PE          ]    };
+  key <AD11> {   [ Cyrillic_ya,         Cyrillic_YA          ]    };
+  key <AD12> {   [ Cyrillic_shcha,      Cyrillic_SHCHA       ]    };
+
+  key <LatA> {   [ Cyrillic_a,          Cyrillic_A           ]    };
+  key <LatS> {   [ Cyrillic_es,         Cyrillic_ES          ]    };
+  key <LatD> {   [ Cyrillic_de,         Cyrillic_DE          ]    };
+  key <LatF> {   [ Cyrillic_ef,         Cyrillic_EF          ]    };
+  key <LatG> {   [ Cyrillic_ghe,        Cyrillic_GHE         ]    };
+  key <LatH> {   [ Cyrillic_ha,         Cyrillic_HA          ]    };
+  key <LatJ> {   [ Cyrillic_shorti,     Cyrillic_SHORTI      ]    };
+  key <LatK> {   [ Cyrillic_ka,         Cyrillic_KA          ]    };
+  key <LatL> {   [ Cyrillic_el,         Cyrillic_EL          ]    };
+key.type[Group1] = "TWO_LEVEL";
+  key <AC10> {   [ semicolon,           colon                ]    };
+  key <AC11> {   [ apostrophe,          quotedbl             ]    };
+key.type[Group1] = "ALPHABETIC";
+  key <BKSL> {   [ Cyrillic_softsign,   U045D                ]    };
+
+  key <LatZ> {   [ Cyrillic_ze,         Cyrillic_ZE          ]    };
+  key <LatX> {   [ Cyrillic_zhe,        Cyrillic_ZHE         ]    };
+  key <LatC> {   [ Cyrillic_tse,        Cyrillic_TSE         ]    };
+  key <LatV> {   [ Cyrillic_ve,         Cyrillic_VE          ]    };
+  key <LatB> {   [ Cyrillic_be,         Cyrillic_BE          ]    };
+  key <LatN> {   [ Cyrillic_en,         Cyrillic_EN          ]    };
+  key <LatM> {   [ Cyrillic_em,         Cyrillic_EM          ]    };
+key.type[Group1] = "TWO_LEVEL";
+  key <AB08> {   [ comma,               doublelowquotemark   ]    };
+  key <AB09> {   [ period,              leftdoublequotemark  ]    };
+  key <AB10> {   [ slash,               question             ]    };
+
+  key <LSGT> {   [ Cyrillic_softsign,   Cyrillic_SOFTSIGN    ]    };
+
+  // End alphanumeric section
+
+  key <KPDL> {   type[Group1] = "KEYPAD",
+                 [ KP_Delete,           KP_Separator         ]    };
+};
+
-- 
1.6.0.4

